src/main.rs

```diff
@@ -21,14 +23,20 @@ fn main() -> anyhow::Result<()> {
 
     let mut mmu = Mmu::new();
     mmu.load_rom(rom);
-    let mut cpu = Cpu::new(mmu);
+    // Multiple mutable references, not allowed.
+    // Need to think about how we should design this.
+    // Makes sense that ppu should instead own VRAM, but MMU should
+    // still be able to write to VRAM, so MMU would need a mut ref to PPU...
+    let mut cpu = Cpu::new(&mut mmu);
```

Comment:
to refcell or not to refcell, that is the question - alternatively the ppu could own vram and maybe get LY, STAT, etc. passed to it which makes sense i guess... another solution is to not keep a mutable reference to the mmu in the structs, but pass to `step` `execute_instruction` when needed